x-pack/qa/reindex-tests-with-security/src/yamlRestTest/java/org/elasticsearch/xpack/security/ReindexWithSecurityIT.java

```diff
@@ -179,24 +188,43 @@ private void createIndicesWithRandomAliases(String... indices) throws IOExceptio
                     request.addAliasAction(AliasActions.add().index(index).alias("alias"));
                 }
             }
-            Request restRequest = new Request("POST", "/_aliases");
+            Request restRequest = new Request(HttpPost.METHOD_NAME, "/_aliases");
             XContentBuilder builder = jsonBuilder();
             request.toXContent(builder, null);
             restRequest.setEntity(new StringEntity(Strings.toString(builder), ContentType.APPLICATION_JSON));
-            Response restResponse = restClient.getLowLevelClient().performRequest(restRequest);
+            Response restResponse = client().performRequest(restRequest);
             AcknowledgedResponse response = AcknowledgedResponse.fromXContent(responseAsParser(restResponse));
             assertThat(response.isAcknowledged(), is(true));
         }
 
         for (String index : indices) {
-            restClient.index(new IndexRequest(index).source("field", "value"), RequestOptions.DEFAULT);
+            execute(HttpPost.METHOD_NAME, index + "/_doc", """
+                {"field":"value"}
+                """, Map.of());
+        }
+        refresh(client(), String.join(",", indices));
+    }
+
+    private static Response execute(String method, String endpoint, String jsonBody, Map<String, String> parameters) throws IOException {
+        Request request = new Request(method, endpoint);
+        request.setJsonEntity(jsonBody);
+        if (parameters != null && parameters.size() > 0) {
+            request.addParameters(parameters);
         }
-        refresh(restClient.getLowLevelClient(), String.join(",", indices));
+        request.setOptions(RequestOptions.DEFAULT);
+        return client().performRequest(request);
     }
 
-    private class TestRestHighLevelClient extends RestHighLevelClient {
-        TestRestHighLevelClient() {
-            super(client(), restClient -> {}, Collections.emptyList());
+    private static Response executeIgnoringConflicts(String method, String endpoint, String jsonBody, Map<String, String> parameters)
```

Comment:
Why is it necessary to ignore 409's in most of the places this is used, given we didn't do that before it looks like?

Reply:
These tests are failing if 409 status is not ignored. I simply replicated the behaviour of RestHighLevelClient. Both [updateByQuery](https://github.com/elastic/elasticsearch/blob/0799663a589aec510877272d41044f276e22fd2d/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L388) and [deleteByQuery](https://github.com/elastic/elasticsearch/blob/0799663a589aec510877272d41044f276e22fd2d/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L406) ignore 409 status code and simply return a response without throwing an exception. 

https://github.com/elastic/elasticsearch/blob/0799663a589aec510877272d41044f276e22fd2d/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L394